app/models/ticket_stakeholder.rb

```diff
@@ -0,0 +1,33 @@
+# frozen_string_literal: true
+
+class TicketStakeholder < ApplicationRecord
+  enum :stakeholder_type, {
+    user: "user",
+    user_group: "user_group",
+  }
+
+  enum :connection, {
+    assigned: "assigned",
+    cc: "cc",
+  }
+
+  belongs_to :ticket
+
+  def stakeholder
+    if stakeholder_type == TicketStakeholder.stakeholder_types[:user_group]
+      UserGroup.find(stakeholder_id)
+    elsif stakeholder_type == TicketStakeholder.stakeholder_types[:user]
+      User.find(stakeholder_id)
+    end
+  end
```

Comment:
Rails has `polymorphic` associations for exactly this purpose. You don't need to write this yourself, just use `belongs_to :stakeholder, polymorphic: true` above.